In [ ]:
from datetime import datetime
import locale

# Définir la locale en français
locale.setlocale(locale.LC_TIME, "fr_FR.UTF-8")

DATE_FORMAT = "%d %b %Y"

In [ ]:
from mongo_episode import Episodes
import pandas as pd


def get_episodes():
    episodes = Episodes()
    all_episodes = episodes.episodes
    episodes_df = pd.DataFrame([episode.to_dict() for episode in all_episodes])
    # episodes_df["date"] = episodes_df["date"].dt.strftime("%Y/%m/%d")
    episodes_df["duree (min)"] = (episodes_df["duree"] / 60).round(1)
    episodes_df.drop(
        columns=["url_telechargement", "audio_rel_filename", "type", "duree"],
        inplace=True,
    )
    return episodes_df

In [ ]:
episodes_df = get_episodes()

# doublons

In [ ]:
episodes_df

date                                              titre  \
0   2025-01-26 10:59:39  Les nouveaux romans de Leïla Slimani, Pierre L...   
1   2025-01-12 10:59:39  Les nouvelles pages de Vanessa Springora, Haru...   
2   2024-12-22 09:59:39  Les nouvelles pages de Marc Dugain, Emmanuelle...   
3   2024-12-01 09:59:39  Les nouvelles pages de James Ellroy, Yasmina R...   
4   2024-11-10 09:59:39  La foire du livre de Brive : les romans de Dan...   
..                  ...                                                ...   
201 1961-11-16 00:00:00  Le Masque et les Goncourt : "La Pitié de Dieu"...   
202 1961-11-16 00:00:00  Le Masque et les Goncourt : "L'Etat sauvage" d...   
203 1960-12-15 00:00:00  Le Masque et les Goncourt : "Dieu est né en ex...   
204 1959-11-19 00:00:00  Le Masque et les Goncourt : "Le Dernier des ju...   
205 1958-12-18 00:00:00  Le Masque et les Goncourt : "Saint-Germain ou ...   

                                           description  \
0    durée : 00:47:56 - Le Masque et la Plume - par...   
1    durée : 00:48:41 - Le Masque et la Plume - par...   
2    durée : 00:47:52 - Le Masque et la Plume - par...   
3    durée : 00:47:17 - Le Masque et la Plume - par...   
4    durée : 00:46:03 - Le Masque et la Plume - par...   
..                                                 ...   
201  Le\n journaliste et écrivain Jean Cau répond a...   
202  En\n 1964, les critiques du "Masque et la plum...   
203  En\n 1960, le prix Goncourt est attribué à Vin...   
204  Un\n livre extraordinairement émouvant, "Le De...   
205  En\n 1958, les jurés du Prix Goncourt récompen...   

                                         transcription  duree (min)  
0     France Inter Le masque et la plume Le sommair...         47.9  
1     France Inter Le masque et la plume Premier ma...         48.7  
2     France Inter Le masque et la plume Cinq roman...         47.9  
3     France Inter Le masque et la plume Les nouvea...         47.3  
4     France Inter Le masque et la plume Bienvenue ...         46.0  
..                                                 ...          ...  
201   France Inter Ça a été votre disparition il y ...          9.0  
202   France Inter en effet il y a plusieurs mois e...          5.0  
203   France Inter Qu'il s'agit encore une fois de ...          1.0  
204   France Inter Commençons par le grand cours qu...          8.0  
205   France Inter il est évident que nous l'avons ...          6.0  

[206 rows x 5 columns]

In [ ]:
# Calculer le nombre d'occurrences de chaque date
date_counts = episodes_df["date"].value_counts()

# Filtrer les dates qui apparaissent plus d'une fois
dates_with_multiple_episodes = date_counts[date_counts > 1]
dates_with_multiple_episodes

date
2022-10-02    3
2021-11-28    2
2018-11-25    2
2019-09-15    2
2020-11-01    2
2017-07-02    2
2016-09-11    2
1961-11-16    2
Name: count, dtype: int64

In [ ]:
episodes_df[episodes_df["date"].isin(dates_with_multiple_episodes.index)]

date                                              titre  \
41  2022-10-02  Les nouvelles pages de Jonathan Franzen, Simon...   
42  2022-10-02  Le Masque et les Goncourt : "Veiller sur elle"...   
43  2022-10-02  Le Masque et les Goncourt : "Vivre vite" de Br...   
58  2021-11-28  Le Masque et les Goncourt : "La plus secrète m...   
59  2021-11-28  Que valent les derniers livres de Richard Ford...   
83  2020-11-01  Livres : les nouveaux opus de Ken Follett, Fat...   
84  2020-11-01  Le Masque et les Goncourt : "L'Anomalie" d'Her...   
102 2019-09-15  Cécile Guilbert, Lionel Duroy, Emma Becker, Je...   
103 2019-09-15  Le Masque et les Goncourt : "Tous les hommes n...   
117 2018-11-25  Haruki Murakami, Jane Birkin, Benoît Duteurtre...   
118 2018-11-25  Le Masque et les Goncourt : "Leurs enfants apr...   
142 2017-07-02                       Le masque et la plume livres   
143 2017-07-02  Le Masque et les Goncourt : "L'Ordre du jour" ...   
153 2016-09-11                      Le masque et la plume à Nancy   
154 2016-09-11  Le Masque et les Goncourt : "Chanson douce" de...   
201 1961-11-16  Le Masque et les Goncourt : "La Pitié de Dieu"...   
202 1961-11-16  Le Masque et les Goncourt : "L'Etat sauvage" d...   

                                           description  \
41   Les\n critiques ont-ils apprécié la lecture de...   
42   Trop\n cliché ou roman populaire de qualité? E...   
43   Avec\n "Vivre vite", Brigitte Giraud revient s...   
58   Un\n débat littéraire captivant où les critiqu...   
59   Entre\n "Rien à déclarer" de Richard Ford, "La...   
83   Faut-\nil lire "Le Crépuscule et l’Aube", de K...   
84   Dans\n cette émission, nos intervenants s'atta...   
102  "Ecrits\n stupéfiants" de Cécile Guilbert, "No...   
103  Les\n critiques ne sont pas d'accord à propos ...   
117  Nos\n critiques aiment-ils “Le Meurtre du Comm...   
118  Dans\n cet extrait du Masque et la plume, nos ...   
142  Vernon\n Subutex 3, V. Despentes / Les Jours e...   
143  Dans\n l'émission "Le Masque et la plume", ani...   
153  «Le\n Grand jeu», Céline Minard/«Chanson douce...   
154  Dans\n cet extrait du Masque et la Plume, l'éq...   
201  Le\n journaliste et écrivain Jean Cau répond a...   
202  En\n 1964, les critiques du "Masque et la plum...   

                                         transcription  duree (min)  
41    France Inter Le masque et la plume Bonsoir à ...         54.0  
42    France Inter. Et on commence avec le lauréat ...         10.0  
43    France Inter. Et on commence avec Vivre vite ...          8.0  
58    France Inter. Et on commence avec le prix Gon...         10.0  
59    France Inter Le masque et la plume Musique Mu...         54.0  
83    France Inter Le masque et la plume Bonsoir à ...         52.0  
84    France Inter. Et maintenant le roman qui a dé...         11.0  
102   France Inter Le masque et la plume consacré à...         53.0  
103   France Inter. la prison de Bordeaux, à Montré...          9.0  
117   France Inter Le masque et la plume Bonsoir à ...         54.0  
118   France Inter roman de 430 pages qui raconte l...          9.0  
142   France Inter Le masque et la plume Bonsoir à ...         54.0  
143   France Inter. diplomatiques, plutôt les march...          4.0  
153   France Inter Le masque et la plume. Musique B...         54.0  
154   France Inter Une chanson douce chez Gallimard...          8.0  
201   France Inter Ça a été votre disparition il y ...          9.0  
202   France Inter en effet il y a plusieurs mois e...          5.0

In [ ]:
episodes_df[episodes_df["date"].isin(dates_with_multiple_episodes.index)][
    "date"
].unique()

<DatetimeArray>
['2022-10-02 00:00:00', '2021-11-28 00:00:00', '2020-11-01 00:00:00',
 '2019-09-15 00:00:00', '2018-11-25 00:00:00', '2017-07-02 00:00:00',
 '2016-09-11 00:00:00', '1961-11-16 00:00:00']
Length: 8, dtype: datetime64[ns]

In [ ]:
def print_ligne(row_num: int, episodes_df=episodes_df):
    row = episodes_df.iloc[row_num]
    date = row["date"].strftime(DATE_FORMAT)
    titre = row["titre"]
    duree = row["duree (min)"]
    print(f"[{row_num}] {date} - {titre} ({duree} min)")

In [ ]:
print_ligne(0)

[0] 26 janv. 2025 - Les nouveaux romans de Leïla Slimani, Pierre Lemaître, Jeanne Rivière, Nathalie Azoulai, Milena Agus (47.9 min)


In [ ]:
def print_ligne_from_date(date: str, episodes_df=episodes_df):
    sous_episode_df = episodes_df[episodes_df["date"] == date]
    for index in sous_episode_df.index:
        print_ligne(index)

In [ ]:
print_ligne_from_date("2022-10-02")

[41] 02 oct. 2022 - Les nouvelles pages de Jonathan Franzen, Simon Liberati, Lola Lafon, Brigitte Giraud… (54.0 min)
[42] 02 oct. 2022 - Le Masque et les Goncourt : "Veiller sur elle" de Jean-Baptiste Andréa, 2023 (10.0 min)
[43] 02 oct. 2022 - Le Masque et les Goncourt : "Vivre vite" de Brigitte Giraud, 2022 (8.0 min)


## 2022-10-02

In [ ]:
print_ligne_from_date("2022-10-02")

[41] 02 oct. 2022 - Les nouvelles pages de Jonathan Franzen, Simon Liberati, Lola Lafon, Brigitte Giraud… (54.0 min)
[42] 02 oct. 2022 - Le Masque et les Goncourt : "Veiller sur elle" de Jean-Baptiste Andréa, 2023 (10.0 min)
[43] 02 oct. 2022 - Le Masque et les Goncourt : "Vivre vite" de Brigitte Giraud, 2022 (8.0 min)


[42] erreur de date, je ne connais pas la date de diffusion mais le goncourt a ete decerne le 7 novembre 2023

[43] date goncourt - 03 oct. 2022

## 2021-11-28

In [ ]:
print_ligne_from_date("2021-11-28")

[58] 28 nov. 2021 - Le Masque et les Goncourt : "La plus secrète mémoire des hommes" de Mohamed Mbougar Sarr, 2021 (10.0 min)
[59] 28 nov. 2021 - Que valent les derniers livres de Richard Ford, Anne Berest, Jean-Christophe Rufin, Mohamed Mbougar Sarr… ? (54.0 min)


[58] 29 nov. 2021


## 2020-11-01

In [ ]:
print_ligne_from_date("2020-11-01")

[83] 01 nov. 2020 - Livres : les nouveaux opus de Ken Follett, Fatima Daas, Julian Barnes, Hervé Le Tellier, Muriel Barbery (52.0 min)
[84] 01 nov. 2020 - Le Masque et les Goncourt : "L'Anomalie" d'Hervé Le Tellier, 2020 (11.0 min)


[84] 02 nov. 2020

## 2019-09-15

In [ ]:
print_ligne_from_date("2019-09-15")

[102] 15 sept. 2019 - Cécile Guilbert, Lionel Duroy, Emma Becker, Jean-Paul Dubois et Jonathan Coe valent-ils la peine d'être lus ? (53.0 min)
[103] 15 sept. 2019 - Le Masque et les Goncourt : "Tous les hommes n'habitent pas le monde de la même façon" de Jean-Paul Dubois, 2019 (9.0 min)


[103] 16 sept. 2019

## 2018-11-25

In [ ]:
print_ligne_from_date("2018-11-25")

[117] 25 nov. 2018 - Haruki Murakami, Jane Birkin, Benoît Duteurtre, Nicolas Mathieu… Quels sont les auteurs à ne pas manquer ? (54.0 min)
[118] 25 nov. 2018 - Le Masque et les Goncourt : "Leurs enfants après eux" de Nicolas Mathieu, 2018 (9.0 min)


[118] - 26 nov. 2018

## 2017-07-02

In [ ]:
print_ligne_from_date("2017-07-02")

[142] 02 juil. 2017 - Le masque et la plume livres (54.0 min)
[143] 02 juil. 2017 - Le Masque et les Goncourt : "L'Ordre du jour" d'Eric Vuillard, 2017 (4.0 min)


[143] - 03 juil. 2017

## 2016-09-11

In [ ]:
print_ligne_from_date("2016-09-11")

[153] 11 sept. 2016 - Le masque et la plume à Nancy (54.0 min)
[154] 11 sept. 2016 - Le Masque et les Goncourt : "Chanson douce" de Leïla Slimani, 2016 (8.0 min)


[154] - 12 sept. 2016

## 1961-11-16

In [ ]:
print_ligne_from_date("1961-11-16")

[201] 16 nov. 1961 - Le Masque et les Goncourt : "La Pitié de Dieu" de Jean Cau, 1961 (9.0 min)
[202] 16 nov. 1961 - Le Masque et les Goncourt : "L'Etat sauvage" de Georges Conchon, 1964 (5.0 min)


[202] - date a changer en 18 nov. 1964

# actions

```text
[42] - date a changer en 07 nov. 2023
[43] - date a changer en 03 oct. 2022
[58] - date a changer en 29 nov. 2021
[84] - date a changer en 02 nov. 2020
[103] - date a changer en 16 sept. 2019
[118] - date a changer en 26 nov. 2018
[143] - date a changer en 03 juil. 2017
[154] - date a changer en 12 sept. 2016
[202] - date a changer en 18 nov. 1964
```


# correction

faire une fonction qui prend en entree la ligne et la nouvelle date

et qui retrouve l'entree DB avec la ligne en passant date et titre, et modifie la date et sauve en db